kcmdline.c

```diff
@@ -0,0 +1,89 @@
+#include "kcmdline.h"
+
+static const CHAR8 allowed[6][12] = {
+	"^console=",
+	"^root=oci:",
+	"root=atomix",
+	"ro",
+	"quiet",
+	"verbose"
+};
+
+#define MAX_TOKENS 128
+
+BOOLEAN is_allowed(const CHAR8 *input) {
+	int len = 0;
+	const CHAR8 * token;
+    int num_allowed = sizeof(allowed) / sizeof(allowed[0]);
+	for (int i=0; i<num_allowed; i++) {
+		token = allowed[i];
+		len = strlena(token);
+		if (token[0] == '^') {
+			len = len - 1;
+			token = &token[1];
+		}
+		if (strncmpa(token, input, len) == 0) {
```

Comment:
Hm, I'm confused here - didn't you want to ensure that, if it did not start with ^, the length of token and input were the same?  I.e. "root" does not match "root=ab" ?

Reply:
hm.. you might be right.

Reply:
thank you. you caught a bug. 
I added this test case that showed failure, and then fixed it.

 {EFI_SECURITY_VIOLATION, "quieter"}